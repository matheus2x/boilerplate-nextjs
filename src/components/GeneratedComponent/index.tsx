import * as S from './styles';

const GeneratedComponent = ({ title = 'GeneratedComponent' }) => (
  <S.Wrapper>
    <h1>{title}</h1>
  </S.Wrapper>
);

export default GeneratedComponent;
