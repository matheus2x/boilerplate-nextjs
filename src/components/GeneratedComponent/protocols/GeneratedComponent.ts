export type GeneratedComponent = {
  title?: string;
};
