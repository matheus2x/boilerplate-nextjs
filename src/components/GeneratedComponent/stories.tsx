import { Meta, StoryObj } from '@storybook/react';
import GeneratedComponent from '.';

export default {
  title: 'components/GeneratedComponent',
  component: GeneratedComponent,
} as Meta;

export const Default: StoryObj = {};
