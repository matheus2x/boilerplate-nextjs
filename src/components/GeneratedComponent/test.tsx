import { render, screen } from '@testing-library/react';

import GeneratedComponent from '.';

describe('<GeneratedComponent />', () => {
  it('should render the heading', () => {
    const { container } = render(<GeneratedComponent />);

    expect(
      screen.getByRole('heading', { name: /GeneratedComponent/i })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
