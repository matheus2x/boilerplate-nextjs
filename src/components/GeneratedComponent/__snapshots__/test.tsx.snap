// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GeneratedComponent /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    GeneratedComponent
  </h1>
</main>
`;
